docs/decisions/0011_multiple_event_types_per_topic.rst

```diff
@@ -0,0 +1,73 @@
+
+Using the TopicRecordName strategy, schemas are registered by topic and record name. The record name is set from the schema as ``<namespace>.<name>`` (or just ``<name>`` if no namespace is provided). On serializing an event, the serializer looks for a schema registered as ``<topic-name>-<record-name>``.
+For example, given a schema like::
+
+    {
+        "doc": "Avro Event Format for CloudEvents created with openedx_events/schema",
+        "fields": [{"name": "str_field", "type": "string"}]
+        "name": "CloudEvent",
+        "namespace": "my.event.type",
+        "type": "record"
+    }
+
+and the topic ``my-topic``, the serializer will look for a schema registered as ``my-topic-my.event.type.CloudEvent`` and use that to determine if the schema being passed is valid. This means that as long as the namespace is different for the schemas for different event types, the serializer will not error when adding new event types. It will still maintain evolution rules within an event type, however, which is desirable behavior.
+
+Consequences
+************
+* The record names for each event type will need to be distinct from each other. Originally, they were all just "CloudEvent." We needed to add namespaces to ensure that they were unique. Note that this will be necessary regardless of which solution we choose.
+* Any event type will be able to be written to any topic, even for topics where we only want a single event type.
+* Schema evolution rules will be enforced separately for each event type on a topic.
+* The event-bus-kafka consumer will need to be updated to determine the signal from the message headers rather than taking a signal as a passed argument in the management command. This will also be necessary regardless of which solution we choose.
+
+
+Rejected Alternatives
+*********************
+
+Setting the subject name strategy on the topic rather than the client
+=====================================================================
+This option is only available for locally-hosted or Dedicated Confluent Cloud clusters, and thus not a feasible solution for the whole community.
+
+Using a SettingsToggle
+======================
+Usually we would put this sort of feature behind a SettingsToggle so deployers could enable it and disable it as they chose. However, because the subject name strategy is set at the client rather than topic or cluster level, all clients who produce to the same topic(s) must have the same subject name strategy set. While the consequences of having multiple clients with different subject name strategies writing to the same topic are not 100% clear, at the very least it would result in a proliferation of confusing schemas in the registry. 
```

Comment:
I'm not clear yet if this section will stay or not until we discuss on the other comment thread.

**IF** it stays, I wonder if we should:
- Add to the decision that we will **not** provide.a toggle to disable. (may not be necessary)
- Clarify that this is not an alternative to the TopicRecordName Strategy decision, but an implementation detail of that alternative. Maybe this would just move to the decision section altogether, and drop this from here?

Reply:
[fixed] removed